### PR TITLE
feat: Devops 895 - azurerm_function_app_slot resource replacement

### PR DESCRIPTION
**Pull request by fmlisco**
### List of changes

- Migration from the azurerm_function_app_slot resource to  the azurerm_linux_function_app_slot one
- the following variables have been removed since they are not used anymore:
os_type
linux_fx_version
resource_group_name
location
app_service_plan_id     

- the following variables have been renamed by the provider:
version ->functions_extension_version
function_app_name -> function_app_id

- the following variables have been added (they replace linux_fx_version):
*Please note you **only** need to set one of these variables (docker one aside)*
      dotnet_version              = var.dotnet_version
      use_dotnet_isolated_runtime = var.use_dotnet_isolated_runtime
      java_version                = var.java_version
      python_version              = var.python_version
      node_version                = var.node_version
      powershell_core_version     = var.powershell_core_version
      use_custom_runtime          = var.use_custom_runtime
      dynamic "docker" {
        for_each = length(var.docker) > 0 ? [1] : []
        content {
          registry_url      = var.docker.registry_url
          image_name        = var.docker.image_name
          image_tag         = var.docker.image_tag
          registry_username = var.docker.registry_username
          registry_password = var.docker.registry_password
        }
      }

### Motivation and context

Since the resource **azurerm_function_app_slot** has been deprecated in version 3.0 of the AzureRM provider, the newer **azurerm_linux_function_app_slot** resource is used in this module. 

### Type of changes

- [x] Resource replaced because of deprecation
- [ ] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [x] Yes
- [ ] No

### Run checks

The Terraform templates saved in the folder **terraform-azurerm-v3/function_app_slot/tests** have been created to test this module.
- cd terraform-azurerm-v3/function_app_slot/tests/
- terraform init
- terraform plan
- terraform apply

### CODE REVIEW

File: function_app_slot/tests/main.tf [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    app_settings_common = local.function_app.app_settings_common
+    app_settings_1 = {
+    }
+    app_settings_2 = {
+    }
+  }
+}
+
+# #tfsec:ignore:azure-storage-queue-services-logging-enabled:exp:2022-05-01 # already ignored, maybe a bug in tfsec
+module "func_python_slot" {
```
Comment (pasqualedevita) [inline function_app_slot/tests/main.tf:65]: probably here is missing the function_app_slot module

File: function_app_slot/README.md [hunk 2]
```diff
@@ -4,42 +4,12 @@ Module that allows the creation of an Azure function app slot.
...
-  subnet_id = module.funcs_diego_snet.id
-
-  allowed_subnets = [
-    module.funcs_diego_snet.id,
-  ]
-
-  tags = var.tags
-}
-```
+No migration is necessary.
```
Comment (pasqualedevita) [inline function_app_slot/README.md:12]: I think that we can copy the function_app script

File: function_app_slot/tests/versions.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 1.3.0"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.30.0, <= 3.43.0"
```
Comment (pasqualedevita) [inline function_app_slot/tests/versions.tf:7]:
```suggestion
      version = ">= 3.30.0, <= 3.44.1"
```

File: function_app_slot/main.tf [hunk 4]
```diff
@@ -4,28 +4,42 @@ locals {
...
-  resource_group_name        = var.resource_group_name
-  location                   = var.location
-  version                    = var.runtime_version
-  function_app_name          = var.function_app_name
-  app_service_plan_id        = var.app_service_plan_id
+  function_app_id            = var.function_app_id
   storage_account_name       = var.storage_account_name
   storage_account_access_key = var.storage_account_access_key
   https_only                 = var.https_only
-  os_type                    = var.os_type
```
Comment (pasqualedevita) [inline function_app_slot/main.tf:17]: add client_certificate_enabled as in https://github.com/pagopa/terraform-azurerm-v3/blob/main/function_app/main.tf#L239

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: function_app_slot/main.tf (hunk 4 above)
Comment (fmlisco) [reply function_app_slot/main.tf:17]: done

File: function_app_slot/tests/main.tf (hunk 1 above)
Comment (fmlisco) [reply function_app_slot/tests/main.tf:65]: fixed

File: function_app_slot/README.md (hunk 2 above)
Comment (fmlisco) [reply function_app_slot/README.md:12]: fixed

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**